src/detect-flowbits.c

```diff
@@ -207,125 +170,86 @@ int DetectFlowbitMatch (DetectEngineThreadCtx *det_ctx, Packet *p,
             return 0;
     }
 
     return 0;
 }
 
-static int DetectFlowbitParse(const char *str, char *cmd, int cmd_len, char *name,
-    int name_len)
+static int DetectFlowbitParse(DetectEngineCtx *de_ctx,
+        const char *rawstr, void **cdout)
 {
-    int count, rc;
-    size_t pcre2len;
-
-    count = DetectParsePcreExec(&parse_regex, str, 0, 0);
-    if (count != 2 && count != 3) {
-        SCLogError(SC_ERR_PCRE_MATCH,
-            "\"%s\" is not a valid setting for flowbits.", str);
-        return 0;
-    }
+    const char *name = NULL;
+    uint32_t cmd = 0;
+    uint32_t vartype = 0;
+    uint32_t or_list_size = 0;
+    void *cd = rs_xbits_parse(rawstr, 1);
 
-    pcre2len = cmd_len;
-    rc = pcre2_substring_copy_bynumber(parse_regex.match, 1, (PCRE2_UCHAR8 *)cmd, &pcre2len);
-    if (rc < 0) {
-        SCLogError(SC_ERR_PCRE_GET_SUBSTRING, "pcre2_substring_copy_bynumber failed");
-        return 0;
+    if (cd == NULL) {
+        return -1;
     }
-
-    if (count == 3) {
-        pcre2len = name_len;
-        rc = pcre2_substring_copy_bynumber(parse_regex.match, 2, (PCRE2_UCHAR8 *)name, &pcre2len);
-        if (rc < 0) {
-            SCLogError(SC_ERR_PCRE_GET_SUBSTRING, "pcre2_substring_copy_bynumber failed");
+    name = rs_xbits_get_name(cd);
+    cmd = rs_xbits_get_cmd(cd);
+    vartype = rs_xbits_get_vartype(cd);
+    or_list_size = rs_xbits_get_or_list_size(cd);
+
+    switch (cmd) {
+        case DETECT_BITS_CMD_NOALERT: {
+            if (name != NULL) {
+                SCLogNotice("Freed cd bc noalert");
+            //    rs_bits_free(cd);
+                return -1;
+            }
+            /* return ok, cd is NULL. Flag sig. */
+            //*cdout = NULL;
             return 0;
         }
-
-        /* Trim trailing whitespace. */
-        while (strlen(name) > 0 && isblank(name[strlen(name) - 1])) {
-            name[strlen(name) - 1] = '\0';
-        }
-
-        if (strchr(name, '|') == NULL) {
-            /* Validate name, spaces are not allowed. */
-            for (size_t i = 0; i < strlen(name); i++) {
-                if (isblank(name[i])) {
-                    SCLogError(SC_ERR_INVALID_SIGNATURE,
-                        "spaces not allowed in flowbit names");
-                    return 0;
-                }
+        case DETECT_BITS_CMD_ISNOTSET:
+        case DETECT_BITS_CMD_ISSET:
+        case DETECT_BITS_CMD_SET:
+        case DETECT_BITS_CMD_UNSET:
+        case DETECT_BITS_CMD_TOGGLE:
+        default:
+            if (strlen(name) == 0) {
+                SCLogNotice("Freed cd bc name");
+              //  rs_bits_free(cd);
+                return -1;
             }
-        }
+            break;
     }
 
-    return 1;
+    if (or_list_size != 0) {
+        for (uint32_t i = 0; i < or_list_size; i++) {
+            int leq = VarNameStoreSetupAdd(rs_xbits_get_or_list_item(cd, i), VAR_TYPE_FLOW_BIT);
+            rs_xbits_set_or_list_id_at(cd, i, leq);
```

Comment:
@jasonish need a little help here.. Made all the args `const void *` to send over to rust and to keep compatibility with the C fn calls but I also want to modify certain fields in Rust on the struct.
Here, I get the error for discarding const qualifier bc I try to get it as mut on the Rust side to modify it.

Reply:
I think the answer is to not pass them as `const`, which might require API changes elsewhere as well.